### DISCUSSION TITLE
aws ecr authentication

### DISCUSSION THREAD

**Discussion by seraphniu** [aquasecurity/trivy#5252] [Q&A]
### Question

Dear team,

I'm trying to use trivy to scan AWS ECR images, but cannot follow instructions from https://aquasecurity.github.io/trivy/v0.17.2/private-registries/ecr/

I setup AWS_ACCESS_KEY_ID /  AWS_SECRET_ACCESS_KEY / AWS_DEFAULT_REGION as env vars, but trivy still reported auth error.

==============================
Then I did below  to make it work:
- install aws cli
- install docker
- setup config(for aws profile) and credentials(key id and secret) under ~/.aws folder.
- use below command for docker to login to AWS ECR.
`aws ecr get-login-password | docker login -u AWS --password-stdin https://111111.dkr.ecr.us-east-1.amazonaws.com`

after all above steps, finally trivy was able to access and scan the images.

however per [doc](https://aquasecurity.github.io/trivy/v0.17.2/private-registries/ecr/), only env vars should be sufficient, those steps(installing aws cli etc) are not required.

Can some one kindly suggest how to make it work without aws cli installation?  also is docker  login required?

Thanks and Regards

### Target

Container Image

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

### Operating System

macos

### Version

```bash
0.45.1
```

**Comment by simar7** [comment]
Please see the update page for private registries https://aquasecurity.github.io/trivy/v0.45/docs/advanced/private-registries/ecr/

**Comment by seraphniu** [comment]
Thank you!  Sorry for the additional questions.  
- Where can I find document on “trivy's service account annotation”?  Is it in a config file?
- What are the mandatory minimal envs? Are keyid +secret enough?

**Reply by seraphniu** [reply]
I saw the service account setup, seems it is more complicate setting than awscli.   Does it support multiple arn for same service account?
